asdf/type_index.py

```diff
@@ -163,17 +166,17 @@ def from_custom_type(self, custom_type):
                 # includes classes that are created dynamically post
```

Comment:
Like what exactly? There is no longer a _CompoundModel class, and no current subclasses of CompoundModel that I'm aware of. Or is this just to handle older versions of astropy?

Reply:
See my comment above, but I suspect the problem this was intended to solve was an import order issue.  Without this feature, the following would succeed:

```python
import SomeCustomCompoundModelSubclass
import asdf.type_index.TypeIndex

type_index = TypeIndex(...)

assert type_index.from_custom_type(SomeCustomCompoundModelSubclass) is not None
```

but this would fail:

```python
import asdf.type_index.TypeIndex

type_index = TypeIndex(...)

import SomeCustomCompoundModelSubclass

assert type_index.from_custom_type(SomeCustomCompoundModelSubclass) is not None
```